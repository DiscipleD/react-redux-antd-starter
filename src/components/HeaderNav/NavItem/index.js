/**
 * Created by jack on 16-9-23.
 */

import React from 'react';
import { Link } from 'react-router';

class NavItem extends React.Component {
	render() {
		const item = this.props.item;
		const navItemElement = item.link
			? <Link to={item.link}>{item.label}</Link>
				: typeof item.fn === 'function'
				? <a onClick={item.fn}>{item.label}</a>
				: item.label;
		return (
			<li className="nav-item">{navItemElement}</li>
		)
	}
}

export default NavItem;
