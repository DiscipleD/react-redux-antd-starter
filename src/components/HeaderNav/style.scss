.header-nav {
  padding: 0 50px;
  background: #404040;
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;

  .header-nav-logo {
    text-align: center;
    padding: 2px 6px;
    font-size: 18px;
    color: #fff;
    width: 120px;
    height: 32px;
    background: #333;
    border-radius: 6px;
  }

  .user-actions{
  }
}
