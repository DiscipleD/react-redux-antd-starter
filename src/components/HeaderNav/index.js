/**
 * Created by jack on 16-9-6.
 */

import React from 'react';

import './style.scss';
import NavItem from './NavItem';

class HeaderNav extends React.Component {
	render() {
		const { logo, navItems: navItemsSetting } = this.props;
		const navItems = (navItemsSetting || []).map((item, key) =>
			<NavItem item={item} key={key} />
		);
		return (
			<div className="header-nav">
				<div className="header-nav-logo" onClick={logo.onClick}>{logo.label}</div>
				<ul className="user-actions">
					{navItems}
				</ul>
			</div>
		)
	}
}

export default HeaderNav;
