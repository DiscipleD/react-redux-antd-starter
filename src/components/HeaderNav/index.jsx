/**
 * Created by jack on 16-9-6.
 */

import React from 'react';
import {Link} from 'react-router';
import {Menu} from 'antd';

import './style.scss';
export default class HeaderNav extends React.Component {
	render() {
		return (
			<div className="header-nav">
				<div className="header-nav-wrapper">
					<div className="header-nav-logo">Project Logo</div>
					<Menu
						theme="dark"
						mode="horizontal"
						defaultSelectedKeys={[this.props.path === '/' ? '/a' : this.props.path]}
						style={{lineHeight: '64px'}}
					>
						<Menu.Item key="/a">
							<Link to="/a">Page A</Link>
						</Menu.Item>
						<Menu.Item key="/b">
							<Link to="/b">Page B</Link>
						</Menu.Item>
					</Menu>
				</div>
			</div>
		)
	}
}
