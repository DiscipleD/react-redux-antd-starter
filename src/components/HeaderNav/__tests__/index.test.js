/**
 * Created by jack on 16-9-18.
 */

import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestUtils from 'react-addons-test-utils';
// import renderer from 'react-test-renderer';
import HeaderNav from '../index';

describe('HeaderNav Component', () => {
	let headerNavNode;
	const menuSetting = [{
		path: '/a',
		name: 'a',
		label: 'Page A'
	}, {
		path: '/b',
		name: 'b',
		label: 'Page B'
	}];
	const selectedItem = menuSetting[1];
	const goHomeFn = jest.fn();

	beforeAll(() => {
		const headerNav = ReactTestUtils.renderIntoDocument(
			<HeaderNav
				navList={menuSetting}
				path={selectedItem.path}
				onlogoClickFn={goHomeFn}
			/>
		);

		headerNavNode = ReactDOM.findDOMNode(headerNav);
	});

	it('Init component nav link according to props', () => {
		expect(headerNavNode.querySelectorAll('li.ant-menu-item').length).toBe(menuSetting.length);

		expect(headerNavNode.querySelector('li.ant-menu-item.ant-menu-item-selected a').textContent).toBe(selectedItem.label);
	});

	// react-dom & react-test-renderer couldn't be import in a file
	// that will be solved in 15.4.0. ref: https://github.com/facebook/react/issues/7386
	/* it('match snapshot', () => {
		const headerNavSnapshot = renderer.create(
			<HeaderNav
				navList={menuSetting}
				path={selectedItem.path}
				onlogoClickFn={goHomeFn}
			/>
		).toJSON();
		expect(headerNavSnapshot).toMatchSnapshot();
	})*/

	it('Logo click', () => {
		ReactTestUtils.Simulate.click(headerNavNode.querySelector('.header-nav-logo'));
		expect(goHomeFn).toBeCalled();
	});

	it('no params', () => {
		const headerNav = ReactTestUtils.renderIntoDocument(
			<HeaderNav />
		);
		const headerNavNode = ReactDOM.findDOMNode(headerNav);

		expect(headerNavNode.querySelectorAll('li.ant-menu-item').length).toBe(0);
		expect(headerNavNode.querySelector('.header-nav-logo')).toBeDefined();
	});
});
