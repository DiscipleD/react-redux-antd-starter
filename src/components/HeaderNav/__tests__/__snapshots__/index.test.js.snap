exports[`HeaderNav Component match snapshot 1`] = `
<div
  className="header-nav">
  <div
    className="header-nav-wrapper">
    <div
      className="header-nav-logo"
      onClick={[Function mockConstructor]}>
      Project Logo
    </div>
    <ul
      aria-activedescendant=""
      className="ant-menu ant-menu-horizontal  ant-menu-dark ant-menu-root"
      onKeyDown={[Function bound onKeyDown]}
      role="menu"
      style={
        Object {
          "lineHeight": "64px"
        }
      }
      tabIndex="0">
      <li
        aria-disabled={undefined}
        aria-selected={false}
        className="ant-menu-item"
        onClick={[Function bound onClick]}
        onMouseEnter={[Function bound onMouseEnter]}
        onMouseLeave={[Function bound onMouseLeave]}
        role="menuitem"
        style={Object {}}
        title={undefined}>
        <a
          onClick={[Function bound handleClick]}
          style={Object {}}>
          Page A
        </a>
      </li>
      <li
        aria-disabled={undefined}
        aria-selected={true}
        className="ant-menu-item-selected ant-menu-item"
        onClick={[Function bound onClick]}
        onMouseEnter={[Function bound onMouseEnter]}
        onMouseLeave={[Function bound onMouseLeave]}
        role="menuitem"
        style={Object {}}
        title={undefined}>
        <a
          onClick={[Function bound handleClick]}
          style={Object {}}>
          Page B
        </a>
      </li>
    </ul>
  </div>
</div>
`;
